TombEngine/Objects/Utils/object_helper.cpp

```diff
@@ -1,146 +1,182 @@
 #include "framework.h"
+#include "Objects/Utils/object_helper.h"
 
 #include "Game/collision/collide_item.h"
 #include "Game/Lara/lara_flare.h"
 #include "Game/pickup/pickup.h"
-#include "Objects/Utils/object_helper.h"
 #include "Objects/Generic/Object/objects.h"
 #include "Objects/Generic/puzzles_keys.h"
 #include "Objects/TR5/Object/tr5_pushableblock.h"
 #include "Specific/level.h"
 
-void InitSmashObject(ObjectInfo* obj, int objectNumber)
+void AssignObjectMeshSwap(ObjectInfo& object, int requiredMeshSwap, const std::string& baseName, const std::string& requiredName)
+{
+	if (Objects[requiredMeshSwap].loaded)
+		object.meshSwapSlot = requiredMeshSwap;
+	else
+		TENLog("Slot " + requiredName + " not loaded. Meshswap issues with " + baseName + " may result in incorrect behaviour.", LogLevel::Warning);
+}
+
+bool AssignObjectAnimations(ObjectInfo& object, int requiredObject, const std::string& baseName, const std::string& requiredName)
+{
+	// Check if the object has at least 1 animation with more than 1 frame.
+	const auto& anim = g_Level.Anims[object.animIndex];
+	if ((anim.frameEnd - anim.frameBase) > 1)
+		return true;
+
+	// Use slot if loaded.
+	if (Objects[requiredObject].loaded)
```

Comment:
As above, you can also make a check here if required object has any animation frames, in case user may accidentally also use a substitute object without animations.

Reply:
fixed